src/web/api/src/main/java/org/locationtech/geogig/rest/repository/UploadCommandResource.java

```diff
@@ -0,0 +1,148 @@
+     * server should POST form data with this value as the key for the file to be uploaded.
+     */
+    public static final String UPLOAD_FILE_KEY = "fileUpload";
+
+    /**
+     * Size, in bytes, of the memory threshold for storing uploaded data. Uploads larger than this
+     * value will be flushed to disk for processing. Uploads smaller than this value will remain in
+     * memory until processed. The threshold is 1MB.
+     */
+    private static final int UPLOAD_THRESHOLD = 0x1000 * 1000;
+
+    private static final MediaType DEFAULT_OUTPUT_MEDIA_TYPE = Variants.XML.getMediaType();
+
+    @Override
+    protected String getCommandName() {
+        // Overriding the getCommandName as this Resource is bound only to "import" context.
+        return "import";
+    }
+
+    @Override
+    protected ParameterSet buildParameterSet(Form options) {
+        // Override the default ParameterSet to include the uploaded file.
+        return new FormParams(options, consumeFileUpload(this.getRequest().getEntity()));
+    }
+
+    @Override
+    protected MediaType resolveFormat(Form options, Variant variant) {
+        if (options.getFirstValue("output_format") != null) {
+            // requested output format exists, use that
+            return super.resolveFormat(options, variant);
+        }
+        // if not requested, the Variant MediaType for uploads is multipart. The response needs to
+        // be XML or JSON, so let's override it
+        return DEFAULT_OUTPUT_MEDIA_TYPE;
+    }
+
+    @Override
+    public void handlePost() {
+        // Request Entity may not be available, as in the case of a file upload.
+        // as long as the entity is not null and at least transient, call post.
+        final Optional<Representation> optional = Optional.fromNullable(getRequest().getEntity());
+        if (optional.isPresent()) {
+            final Representation entity = optional.get();
+            if (entity.isAvailable() || entity.isTransient()) {
+                post(getRequest().getEntity());
+                return;
+            }
+        }
+        // just call super
+        super.handlePost();
+    }
+
+    @Override
+    public void post(Representation entity) {
+        // invoke the runCommand method on the parent and get the representation
+        // set the representation on the response
+        getResponse().setEntity(runCommand(entity, getRequest()));
+    }
+
+    /**
+     * Consumes the data sent from the client and stores it into a temporary file to be processed.
+     * This method is just looking through the request entity for form data named
+     * {@value #UPLOAD_FILE_KEY}. If present, we will consume the data stream from the request and
+     * store it in a temporary file.
+     *
+     * @param entity POSTed entity containing binary data to be processed.
+     *
+     * @return local File representation of the data streamed form the client.
+     */
+    private File consumeFileUpload(Representation entity) {
```

Comment:
@groldan I changed the implementation of `consumeFileUpload` to return `null` if the request entity does not have exactly 1 `fileUpload`. At this point in the code, it's too early to use `CommandSpecException` as it will blow up too soon and give a generic 500 error.
Inside the command, if `fileUpload` is `null`, it throws a `CommandSpecException` indicating 1 and only 1 `fileUpload` must be in the request.